qa/workunits/fs/fscrypt_protect.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+set -xe
+
+client_type=$1
+
+case ${client_type} in
+	"kclient")
+		# test contents protection
+		echo "hello" > fscrypt_test_file
+		touch ../fscrypt_kclient_ready
+		while [ -f ../fscrypt_kclient_ready ]
+		do
+			sleep 5
+		done
+		if grep -q "hello" fscrypt_test_file; then
+			echo "Fscrypt contents protect test successfully!"
+		else
+			echo "Fscrypt contents protect test failed!"
+			exit 1
+		fi
+		rm -f fscrypt_test_file
+
+		# test creating directory dentries, the fuse client should
+		# fail and there should be no new dentries to be created
+		mkdir fscrypt_dir
+		touch ../fscrypt_kclient_ready
+		while [ -f ../fscrypt_kclient_ready ]
+		do
+			sleep 5
+		done
+		num=$(ls -l fscrypt_dir/ | wc -l)
+		if [ $num -eq 1 ]; then
+			echo "Fscrypt create dir dentries protect test successfully!"
+		else
+			echo "Fscrypt create dir dentries protect test failed!"
+			exit 1
+		fi
+		touch ../fscrypt_kclient_ready
+
+		;;
+	"fuse")
+		while [ ! -f ../fscrypt_kclient_ready ]
+		do
+			sleep 5
+		done
+		set +e
+		echo > ./*
+		set -e
+		rm -f ../fscrypt_kclient_ready
+
+		while [ ! -f ../fscrypt_kclient_ready ]
+		do
+			sleep 5
+		done
+		set +e
+		touch `ls`/fscrypt_test_file
+		mkdir `ls`/fscrypt_subdir
+		ln -s . `ls`/fscrypt_symlink
+		set -e
```

Comment:
> The ceph-fuse client couldn't write to or truncate the encrypted file, and also couldn't create files in a encrypted directory too

Can you ensure this section of test fails if any ret code of 0 is returned? I'm concerned about false "successes" i.e any of those commands succeed.

Reply:
It doesn't matter and we do not care the return values here.

All the above `touch/mkdir/ln` certainly should fail, so this is why I do `set +e` before it. And then after this in the `kclinet` side we will check whether the total dentry number has changed in the encrypted dir. The expected result is false, else the test will fail.